plugins/experimental/statichit/statichit.cc

```diff
@@ -62,21 +63,80 @@ static int StaticHitInterceptHook(TSCont contp, TSEvent event, void *edata);
 static int StaticHitTxnHook(TSCont contp, TSEvent event, void *edata);
 
 struct StaticHitConfig {
-  explicit StaticHitConfig(const std::string &filePath, const std::string &mimeType, bool disableExact)
-    : filePath(filePath), mimeType(mimeType), disableExact(disableExact)
+  explicit StaticHitConfig(const std::string &fileOrDir, const std::string &mimeType, bool exact) : mimeType(mimeType)
   {
+    std::filesystem::path base_path;
+
+    if (fileOrDir.find('/') != 0) {
+      base_path = std::filesystem::weakly_canonical(std::string(TSConfigDirGet()) + "/" + fileOrDir);
+    } else {
+      base_path = std::filesystem::weakly_canonical(fileOrDir);
+    }
+
+    if (std::filesystem::is_directory(base_path)) {
+      dirPath      = base_path;
+      filePath     = "";
+      disableExact = true;
+    } else if (std::filesystem::is_regular_file(base_path)) {
+      dirPath      = "";
+      filePath     = base_path;
+      disableExact = exact;
+    } else {
+      VERROR("Invalid file path: %s", filePath.c_str());
+      filePath = "";
+      dirPath  = "";
+    }
   }
 
   ~StaticHitConfig() { TSContDestroy(cont); }
 
+  std::string_view
+  makePath(TSHttpTxn txnp, std::string &output) const
+  {
+    if (!dirPath.empty()) {
```

Comment:
If this is a member variable, you need to either use leading underscores or `this->dirPath` so it's clear where the value is from.

Reply:
Just staying with the way the original plugin was written, where public member variables are not underscored, and we do not explicitly specify this->